keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1116 @@
+      - namespaceSetAndPodSelector:
+          scope: Selector 
+          namespaceSelector: 
+            matchLabels:
+              kubernetes.io/metadata.name: bar-ns-1 
+          podSelector: 
+            matchLabels:
+              app: svc-pub
+```
+
+#### Sample spec for Story 4: Create and Isolate multiple tenants in a cluster
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: tenant-creation-example
+spec:
+  priority: 50
+  - appliedTo:
+     namespaceSelector:
+       matchExpressions: {key: "tenant"; operator: Exists}
+    ingress:
+     - action: Deny
+       from:
+       - namespaceSetSelector:
+          scope: NotSameLabels
+          labels:
+            - tenant
+```
+
+Note: the above AdminNetworkPolicy can also be written in the following fashion:
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: tenant-creation-example
+spec:
+  priority: 50
+  - appliedTo:
+     namespaceSelector:
+       matchExpressions: {key: "tenant"; operator: Exists}
+    ingress:
+     - action: Pass
+       from:
+       - namespaceSetSelector:
+          scope: SameLabels
+          labels:
+            - tenant
+     - action: Deny   # Deny everything else other than same tenant traffic
+       from:
+       - namespaceSetSelector:
+          scope: Selector
+          namespaceSelector: {}
+```
+The difference is that in the first case, traffic within tenant Namespaces will fall
+through, and be evaluated against lower-priority ClusterNetworkPolicies, and then
+NetworkPolicies. In the second case, the matching packet will skip all AdminNetworkPolicy
+evaluation (except for AdminNetworkPolicy priority=baseline), and only match
+against NetworkPolicy rules in the cluster. In other words, the second AdminNetworkPolicy
+specifies intra-tenant traffic must be delegated to the tenant Namespace owners.
+
+#### Sample spec for Story 5: Cluster Wide Default Guardrails
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata: 
+  name: baseline-rule-example
+spec:
+  priority: 0
+  - appliedTo:
+     namespaceSelector: {}
+    ingress:
+     - action: Deny   # zero-trust cluster default security posture
+       from:
+       - namespaceSetSelector:
+          scope: Selector 
+          namespaceSelector: {}
+```
+
+### Test Plan
+
+- Add e2e tests for AdminNetworkPolicy resource
+  - Ensure `Pass` rules are delegated and are not subject to ANP rules.
+  - Ensure `Deny` rules drop traffic.
+  - Ensure `Allow` rules allow traffic.
+  - Ensure that in stacked ClusterNetworkPolicies/K8s NetworkPolicies, precedence is maintained
+    as per the `priority` set in ANP.
+- e2e test cases must cover ingress and egress rules
+- e2e test cases must cover port-ranges, named ports, integer ports etc
+- e2e test cases must cover various combinations of `appliedTos` and `namespaceSet*s` in each ingress/egress rule
```

Comment:
test for overlapping rules and verify priorities and conflict resoultion?

Reply:
this is already here, duh
  - Ensure that in stacked ClusterNetworkPolicies/K8s NetworkPolicies, precedence is maintained
    as per the `priority` set in ANP.